From 05050b84320e118fc7490609b77b5cdeafd83add Mon Sep 17 00:00:00 2001
From: mlevitt <mark.s.levitt@gmail.com>
Date: Sat, 14 Apr 2018 14:51:22 -0600
Subject: [PATCH] (FACT-1844) JDK10 drops support for javah

Any JDK post 9 requires change of invocation from

   javah -d <dir>
to
   javac -h <dir>
---
 lib/CMakeLists.txt | 12 +++++++++++-
 1 file changed, 11 insertions(+), 1 deletion(-)

diff --git a/lib/CMakeLists.txt b/lib/CMakeLists.txt
index be67e0e2b..f7e8d4493 100644
--- a/lib/CMakeLists.txt
+++ b/lib/CMakeLists.txt
@@ -283,8 +283,18 @@ if (JRUBY_SUPPORT)
 
     # javah does not atomically write the header file, so parallel builds can
     # read it before it finishes writing if not careful.
+    # JDK versions after 9 don't provide javah. Use javac in these cases
+
+    if(Java_VERSION VERSION_LESS "10")
+	    set(JAVAH_COMMAND javah)
+	    set(JAVAH_ARG -classpath facter.jar -d "${CMAKE_CURRENT_LIST_DIR}/src/java" com.puppetlabs.Facter)
+    else()
+	    set(JAVAH_COMMAND javac)
+	    set(JAVAH_ARG -h  "${CMAKE_CURRENT_LIST_DIR}/src/java" com/puppetlabs/Facter.java)
+    endif()
+
     add_custom_command(OUTPUT "${CMAKE_CURRENT_LIST_DIR}/src/java/com_puppetlabs_Facter.h"
-                       COMMAND javah ARGS -classpath facter.jar -d "${CMAKE_CURRENT_LIST_DIR}/src/java" com.puppetlabs.Facter
+                       COMMAND ${JAVAH_COMMAND} ARGS ${JAVAH_ARG}
                        WORKING_DIRECTORY "${CMAKE_BINARY_DIR}/lib"
                        DEPENDS facter-jruby-jar)
     # Anything that depends on facter-jruby wants both the jar AND the completely written header.
